### PR TITLE
Add fork

### PR DESCRIPTION
**Pull request by n1koo**
# What

`fork` + `sed` for items. Useful if you have a dash that you want to keep as base for other things and just change the variable names.

Kept this super simple, essentially just copying existing item and then doing `gsub` on it as a string.

# Usage

`fork 214853 --variables="Remember Me:My App Rocks" remember-me:my-app-rocks CheckoutRememberMe:MyAppRocks`

^ worth noticing I put in zero sanity checking. I first thought of making types for title, template variables and such but that made it quite burdensome and had to put same things in to multiple places.

# Fucky
Somewhat dangerous, but given the context of the tool I feel like if the user wants to shoot himself in the leg by creating non functional new item.. thats kinda fine.

Open to opinions on this though

@bai @Shopify/cloudplatform

### CODE REVIEW

**Comment by n1koo** [comment]
ping, also maybe @ElvinEfendi since you seem to be working on the codebase

**Comment by ElvinEfendi** [comment]
I like the idea of `fork` command. But I would make it simpler by getting rid of `variables` option. I think this is a special case therefore does not worth to be supported as a command argument. The same can be achieved with a slightly different but more flexible process. For example, let's say a user wants to create another dashboard based on dashboard with ID=10, then she executes `doggy fork 10` and a new tab opens in the browser to edit the new dashboard that's been created based on another dashboard(10). During this time the process is on hold - same as what happens with doggy edit. When user is done editing, the process resumes, fetches the new item from Datadog and saves it locally. After that if some `sed`ing is necessary user can do it locally without using doggy. Once user is done making changes to the file locally, she makes a PR and deploys the changes.

File: bin/dev [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path("../../lib", __FILE__))
+
+require "pry"
+require "doggy"
+Doggy::CLI.start(ARGV)
```
Comment (ElvinEfendi) [inline bin/dev]: There's `bin/console` already why is this necessary?

File: bin/dev [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env ruby
+
+$LOAD_PATH.unshift(File.expand_path("../../lib", __FILE__))
```
Comment (ElvinEfendi) [inline bin/dev]: Using `require "bundler/setup"` would be better I think because it takes care of including gems mentioned in Gemfile into `LOAD_PATH` a well.

File: exe/doggy [hunk 3]
```diff
@@ -1,4 +1,6 @@
 #!/usr/bin/env ruby
 
+$LOAD_PATH.unshift(File.expand_path("../../lib", __FILE__))
```
Comment (ElvinEfendi) [inline exe/doggy]: I think this is not necessary for a gem executable. `lib` folder gets added to the PATH anyways.

File: bin/dev (hunk 1 above)
Comment (n1koo) [inline bin/dev]: it starts a console while this gives the same thing as running from `cli` + `pry`

File: exe/doggy (hunk 3 above)
Comment (n1koo) [inline exe/doggy]: Ye this was here by mistake

**Comment by n1koo** [comment]
>  But I would make it simpler by getting rid of variables

I agree if this was going to be used by humans, but the user will be spy doing things automatically

**Comment by n1koo** [comment]
@Shopify/cloudplatform ping

**Comment by ElvinEfendi** [comment]
In https://github.com/Shopify/doggy/pull/31/files I have introduced `Doggy::Model.find_local` method to not duplicate `resource_by_param` method. You can use it for this command as well and get rid if `resource_by_param`.

Also can you add a test please.
